Make J9::ClassEnv::getArrayElementWidthInBytes() work in JITServer

jdmpapin opened the pull request:
Previously it would try to dereference the class on the server.

mpirvu reviewed (commented):
The code works, but I would like to eliminate some potential network messages by taking advantages of the session caches. JITServer caches each ROMClass that is aware of.
I am suggesting a change like this:
```
uintptr_t
J9::ClassEnv::getArrayElementWidthInBytes(TR::Compilation *comp, TR_OpaqueClassBlock* arrayClass)
   {
   TR_ASSERT(TR::Compiler->cls.isClassArray(comp, arrayClass), "Class must be array");
   J9ROMClass *romClass = TR::Compiler->cls.romClassOf(arrayClass);
   int32_t logElementSize = ((J9ROMArrayClass*)romClass)->arrayShape & 0x0000FFFF;
   return 1 << logElementSize;
   }
```
 `TR::Compiler->cls.romClassOf` is JITServer aware.

jdmpapin commented:
Thanks for the suggestion! Updated as requested

Since there is no longer any change to the communication stream protocol, I removed my `MINOR_NUMBER` increment. I'd still like to introduce the increment ID comment I had there, but since it's not related to this change anymore I've opened a separate PR #17292 for that

mpirvu reviewed (approved):
LGTM

mpirvu commented:
jenkins test sanity plinuxjit,xlinuxjit,zlinuxjit jdk17
